app/controllers/channels_controller.rb

```diff
@@ -0,0 +1,78 @@
+class ChannelsController < ApplicationController
+  before_action :set_channel, only: [:show, :edit, :update, :destroy]
+  skip_before_action :verify_authenticity_token
+
+  # GET /channels
+  # GET /channels.json
+  def index
+    @channels = Channel.all
+  end
+
+  # GET /channels/1
+  # GET /channels/1.json
+  def show
+    @channel = Channel.find(params[:id])
+    render json: @channel
+  end
+
+  # GET /channels/new
+  def new
+    @channel = Channelel.new
+  end
+
+  # GET /channels/1/edit
+  def edit
+    @channel = Channel.find(params[:id])
+  end
+
+  # POST /channels
+  # POST /channels.json
+  def create
+    @channel = Channel.new(channel_params)
+
+    respond_to do |format|
+      if @channel.save
+        #format.html { redirect_to @channel, notice: 'Channel was successfully created.' }
```

Comment:
@huutamnguyen we are only implement restful API. So please review these lines in all controllers.

Reply:
ok, it is implemented now